TEMPLATES.md

```diff
@@ -0,0 +1,103 @@
+# Using `Template`s as Execution Targets With CI Operator
+
+The CI Operator allows for users to configure tests running in a single container
+with the `tests` array. In many cases, however, running a command in a single
+container will not suffice. For more complicated tests, the CI Operator supports
+a `--template` option with which a test author may write black-boxed execution
+targets using OpenShift `Template`s. This document explains the interfaces expected
```

Comment:
I'm gonna guess OpenShift Templates are a foreign concept to most readers, so perhaps link to the docs for them here.